pkg/util/chunk/pool.go

```diff
@@ -18,8 +18,44 @@ import (
 	"sync"
 
 	"github.com/pingcap/tidb/pkg/types"
+	"github.com/pingcap/tidb/pkg/util/syncutil"
 )
 
+var (
+	globalChunkPoolMutex syncutil.RWMutex
+	// globalChunkPool is a chunk pool, the key is the init capacity.
+	globalChunkPool = make(map[int]*Pool)
+)
+
+func getChunkFromPool(initCap int, fields []*types.FieldType) *Chunk {
+	globalChunkPoolMutex.RLock()
```

Comment:
Compare with before, is there any performance issue for the destroy and get ?

Reply:
In fact, as long as we don't encounter a new size, we are dealing with a read mutex. But since there are only so many sizes, it would only impact at startup. After that, it should all be read mutex.

Reply:
Additionally, if we really encounter performance issues,  this part can be turned into a shardmap. However, this would increase the code complexity here and affect the review process. We can consider this for subsequent optimizations.

Reply:
BTW， this initCap is the size of bucket in the histogram. I think it will not have too many difference values.